src/metrics.jl

```diff
@@ -182,3 +183,291 @@ function fraction_within(values, bitmask, start, stop)
+                                 class_indices)
+    hardened_metrics_table = vcat(hardened_metrics_table,
+                                  get_hardened_metrics_multiclass(predicted_hard_labels,
+                                                                  elected_hard_labels,
+                                                                  length(classes)))
+
+    # Step 4: Calculate all metrics derived directly from labels (does not depend on
+    # predictions)
+    labels_metrics_table = map(c -> get_label_metrics(votes, c), class_indices)
+    labels_metrics_table = vcat(labels_metrics_table,
+                                get_label_metrics_multiclass(votes, length(classes)))
+
+    # Adendum: Not including `stratified_kappas` by default in any of our metrics
+    # calculations; including here so as not to fail the deprecation sanity-check
+    stratified_kappas = has_value(strata) ?
+                        _calculate_stratified_ea_kappas(predicted_hard_labels,
+                                                        elected_hard_labels,
+                                                        length(classes),
+                                                        strata) : missing
+
+    return EvaluationRow(tradeoff_metrics_rows, hardened_metrics_table,
+                         labels_metrics_table;
+                         optimal_threshold_class, class_labels, thresholds,
+                         optimal_threshold, stratified_kappas)
+end
+
+function _split_classes_from_multiclass(table)
+    table = DataFrame(table; copycols=false)
+
+    # Pull out individual classes
+    class_rows = filter(:class_index => c -> isa(c, Int), table)
+    sort!(class_rows, :class_index)
+    nrow(class_rows) == length(unique(class_rows.class_index)) ||
+        throw(ArgumentError("Multiple rows for same class!"))
+
+    # Pull out multiclass
+    multi_rows = filter(:class_index => ==(:multiclass), table)
+    nrow(multi_rows) > 1 &&
+        throw(ArgumentError("More than one `:multiclass` row in table!"))
+    multi = nrow(multi_rows) == 1 ? only(multi_rows) : missing
+    return class_rows, multi
+end
+
+function _values_or_missing(values)
+    has_value(values) || return missing
+    return all(ismissing, values) ? missing : values
+end
+
+_unpack_curves(curve::Union{Missing,Curve}) = ismissing(curve) ? missing : Tuple(curve)
+_unpack_curves(curves::AbstractVector{Curve}) = Tuple.(curves)
+
+# `EvaluationRow` constructor to facilitate sanity-check refactor
+function Legolas.Row{S}(tradeoff_metrics_table, hardened_metrics_table, label_metrics_table;
+                        optimal_threshold_class=missing, class_labels, thresholds,
+                        optimal_threshold,
+                        stratified_kappas=missing) where {S<:Legolas.Schema{Symbol("lighthouse.evaluation"),
+                                                                            1}}
```

Comment:
I would probably do this as a separate function rather than the constructor to make sure we don't clash with the default constructor somehow
```suggestion
# `EvaluationRow` constructor to facilitate sanity-check refactor
function evaluation_row(tradeoff_metrics_table, hardened_metrics_table, label_metrics_table;
                        optimal_threshold_class=missing, class_labels, thresholds,
                        optimal_threshold,
                        stratified_kappas=missing)
```